Entity/Repository/JobRepository.php

```diff
@@ -152,22 +152,9 @@ public function getOrCreateIfNotExists($command, array $args = array(), $queueNa
         return $firstJob;
     }
 
-    public function findStartableJob($queueName, array &$excludedIds = array())
+    public function findStartableJob($queueName)
     {
-        while (null !== $job = $this->findPendingJob($queueName, $excludedIds)) {
-            if ($job->isStartable()) {
-                return $job;
-            }
-
-            $excludedIds[] = $job->getId();
-
-            // We do not want to have non-startable jobs floating around in
-            // cache as they might be changed by another process. So, better
-            // re-fetch them when they are not excluded anymore.
-            $this->_em->detach($job);
-        }
-
-        return null;
+        return $this->findPendingJob($queueName);
```

Comment:
Right, so if dependencies is empty then what complexities does it add to the query? I guess my main question is: what problem is this change supposed to solve?